core/arch/arm/include/kernel/thread.h

```diff
@@ -308,6 +323,9 @@ void thread_restore_irq(void);
 #define THREAD_EXCP_ABT	(1 << 2)
 #define THREAD_EXCP_ALL	(THREAD_EXCP_FIQ | THREAD_EXCP_IRQ | THREAD_EXCP_ABT)
 
+/* Defines the bit mask for the foreign interrupts */
+#define THREAD_FOREIGN_INTR	THREAD_EXCP_IRQ
```

Comment:
Yes, we can use the code like:
`#ifdef CFG_GICV3
xxxxx(THREAD_EXCP_FIQ)
#else
xxxxx(THREAD_EXCP_IRQ)
#endif
`
in every place using THREAD_EXCP_IRQ.

There are two reasons I defined "THREAD_FOREIGN_INTR" (should be "THREAD_EXCP_FOREIGN_INTR") :)
* The places using "THREAD_EXCP_IRQ" in the current code, actually, just want to mask the interrupt which will cause the exit of optee_os. (e.g. non-secure interrupts). This kind of interrupts are called "foreign" interrupts in this patch. So the name "THREAD_EXCP_FOREIGN_INTR" has clearer meaning than "THREAD_EXCP_IRQ" in the code. And these functions just care about masking foreign interrupts whatever they are sent as FIQ or IRQ. (or even other types in the future)
* We just need to redefine "THREAD_EXCP_FOREIGN_INTR" in one place (thread.h) to support GICv3, like:
`
#ifdef CFG_GICV3
#define THREAD_EXCP_FOREIGN_INTR	THREAD_EXCP_FIQ
#else
#define THREAD_EXCP_FOREIGN_INTR	THREAD_EXCP_IRQ
#endif
`
We don't need to add #ifdef ... #else ... #endif in each place which wants to mask/unmask foreign interrupts.
Actually, "THREAD_EXCP_NATIVE_INTR" could also be defined. But I didn't find any place to use for now in C functions. The places using THREAD_EXCP_FIQ seem to mask all interrupts not specific to "native". (will check again)
Hope this makes sense.